services/processor/native/adapter/native_compiler_test.go

```diff
@@ -23,6 +26,69 @@ import (
 
 const COUNTER_CONTRACT_START_FROM = 100
 
+func TestCompileValidContract(t *testing.T) {
+	with.Logging(t, func(parent *with.LoggingHarness) {
+		if testing.Short() {
+			t.Skip("Skipping compilation of contracts in short mode")
+		}
+		// give the test one minute timeout to compile
+		ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+		defer cancel()
+
+		tmpDir := test.CreateTempDirForTest(t)
+		defer os.RemoveAll(tmpDir)
+		cfg := &adapterTest.HardcodedConfig{ArtifactPath: tmpDir}
+		compiler := NewNativeCompiler(cfg, parent.Logger, metric.NewRegistry())
+
+		code := string(contracts.NativeSourceCodeForCounter(contracts.MOCK_COUNTER_CONTRACT_START_FROM))
+		_, err := compiler.Compile(ctx, code)
+		t.Log("Error : ", err)
+		require.NoError(t, err, "compile should not fail")
+	})
+}
+
+func TestCompileTimeout(t *testing.T) {
+	with.Logging(t, func(parent *with.LoggingHarness) {
+		if testing.Short() {
+			t.Skip("Skipping compilation of contracts in short mode")
+		}
+		// give the test one minute timeout to compile
+		ctx, cancel := context.WithTimeout(context.Background(), 2000*time.Millisecond)
```

Comment:
why 2 seconds? it should be very low value, no?

Reply:
also, the comment above is copy-pasted from previous test, please remove
also, instead of `2000 * time.Millisecond` use `2 * time.Second`

Reply:
> why 2 seconds? it should be very low value, no?
if it's too low, the compile command never starts. this is a best-effort, tested-on-my-machine timing to kill the command  while it executes

Reply:
if the test passes when the process takes longer than 2 seconds to launch then it's fine, otherwise we have a flaky test, no?